website_sale_affiliate/controllers/main.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl)
+import logging
+
+from odoo.http import request, route
+from odoo.addons.website_sale.controllers.main import WebsiteSale as Base
+
+_logger = logging.getLogger(__name__)
+
+
+class WebsiteSale(Base):
+    def _store_affiliate_info(self, **kwargs):
```

Comment:
maybe you can improve simplicity and readability like this:
```python
def _store_affiliate_info(self, **kwargs):
    if 'ref' in kwargs:
        if kwargs['ref'].isdigit():
            request.session['affiliate_id'] = int(kwargs['ref'])
            if 'key' in kwargs:
                request.session['affiliate_key'] = kwargs['key']
        else:
            _logger.debug('Invalid affiliate ID value')
```
BTW maybe is better to provide better naming for those vars, like `aff_key` and `aff_ref`?